### PR TITLE
Improve documentation of derive macros

### PR DESCRIPTION
**Pull request by turion**
This clarifies the usage of the `Nif*` derive macros a bit, and removes a few small errors.

### CODE REVIEW

File: rustler_codegen/src/lib.rs [hunk 1]
```diff
@@ -70,9 +70,9 @@ pub fn nif(args: TokenStream, input: TokenStream) -> TokenStream {
     nif::transcoder_decorator(args, input).into()
 }
 
-/// Implementation of the `NifStruct` macro that lets the user annotate a struct that will
-/// be translated directly from an Elixir struct to a Rust struct. For example, the following
-/// struct, annotated as such:
+/// Create a correspondence between an Elixir struct and a Rust struct.
```
Comment (hansihe) [inline rustler_codegen/src/lib.rs:73]: > Macro for automatically deriving `Encoder` and `Decoder` implementations which convert between a Rust struct and an Elixir struct. 

How about something like this? In my mind the word correspondence might not be the most descriptive in this case.

Comment (hansihe) [reply rustler_codegen/src/lib.rs:73]: The same also applies in the other cases this wording is used.

**Review by hansihe — COMMENTED** [review]
Thanks for the PR!

Generally like the changes and think they make for a much easier to understand documentation, I just have some small nits :)

File: rustler_codegen/src/lib.rs (hunk 1 above)
Comment (turion) [reply rustler_codegen/src/lib.rs]: That's definitely an improvement! I changed the wording further, hopefully keeping your spirit.

**Comment by turion** [comment]
@hansihe What do you think about this version?

**Review by philss — APPROVED** [review]

**Pull request merged**